diff --git a/plugins/include/clients.inc b/plugins/include/clients.inc
index d88262f1..4d4711d4 100644
--- a/plugins/include/clients.inc
+++ b/plugins/include/clients.inc
@@ -69,7 +69,7 @@ enum AuthIdType
  * MAXPLAYERS is a hardcoded value as an upper limit.  MaxClients changes based on the server.
  */
 
-#define MAXPLAYERS      101  /**< Maximum number of players SourceMod supports */
+#define MAXPLAYERS      256  /**< Maximum number of players SourceMod supports */
 #define MAX_NAME_LENGTH 128 /**< Maximum buffer required to store a client name */
 #define MAX_AUTHID_LENGTH 64 /**< Maximum buffer required to store any AuthID type */
 
diff --git a/public/IPlayerHelpers.h b/public/IPlayerHelpers.h
index edb0dc57..d7123773 100644
--- a/public/IPlayerHelpers.h
+++ b/public/IPlayerHelpers.h
@@ -49,7 +49,7 @@ class IPlayerInfo;
 #define SM_REPLY_CONSOLE	0			/**< Reply to console. */
 #define SM_REPLY_CHAT		1			/**< Reply to chat. */
 
-#define SM_MAXPLAYERS				101		/**< Maxplayer Count */
+#define SM_MAXPLAYERS				256		/**< Maxplayer Count */
 
 namespace SourceMod
 {
